### ISSUE TITLE
[Flake] 100ms is too short for thread timeouts

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-cpp#3962]
Looks like another timing problem, last seen in a Windows+Bazel build.

```
INFO: From Testing //google/cloud:internal_background_threads_impl_test:
==================== Test output for //google/cloud:internal_background_threads_impl_test:
Running main() from gmock_main.cc
[==========] Running 3 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 2 tests from CustomerSuppliedBackgroundThreads
[ RUN      ] CustomerSuppliedBackgroundThreads.LifecycleNoShutdown
[       OK ] CustomerSuppliedBackgroundThreads.LifecycleNoShutdown (15 ms)
[ RUN      ] CustomerSuppliedBackgroundThreads.SharesCompletionQueue
[       OK ] CustomerSuppliedBackgroundThreads.SharesCompletionQueue (3 ms)
[----------] 2 tests from CustomerSuppliedBackgroundThreads (18 ms total)

[----------] 1 test from AutomaticallyCreatedBackgroundThreads
[ RUN      ] AutomaticallyCreatedBackgroundThreads.IsActive
google/cloud/internal/background_threads_impl_test.cc(79): error: Expected equality of these values:
  std::future_status::ready
    Which is: 4-byte object <00-00 00-00>
  expired.wait_for(ms(100))
    Which is: 4-byte object <01-00 00-00>
[  FAILED  ] AutomaticallyCreatedBackgroundThreads.IsActive (127 ms)
[----------] 1 test from AutomaticallyCreatedBackgroundThreads (127 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 2 test suites ran. (146 ms total)
[  PASSED  ] 2 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] AutomaticallyCreatedBackgroundThreads.IsActive

 1 FAILED TEST
================================================================================
```

https://source.cloud.google.com/results/invocations/60a6b8d9-0028-46b9-b439-9ed0bb6e5786

**Comment by coryan** [comment]
A very similar one:

https://source.cloud.google.com/results/invocations/3f741ca2-c9b8-4a7a-a083-adc8b8340076

**Comment by coryan** [comment]
I changed the title because it seems we have a lot of tests that do something in one thread and then expect something to happen in another thread within 100ms. That seems too short. If my memory serves me, we had to fix the `future<T>` tests to use 500ms timeouts (see #2047).

Ideally we should avoid these timeouts altogether, and write the tests without them. If we cannot, the 500ms timeouts seem to be Okay in practice (though there is no guarantee, of course).
